frontend-react/src/components/MessageTracker/MessageDetails.test.tsx

```diff
@@ -141,14 +141,15 @@ const MOCK_EMPTY_MESSAGE_DETAIL = {
     errors: [],
     receiverData: [],
 };
-const MOCK_MESSAGE_DETAIL = {
+let MOCK_MESSAGE_DETAIL = {
```

Comment:
We can keep this a `const` and just spread whenever we need to override anything inside:

```
// usually better practice to name this something like "BASE" or "DEFAULT" like BASE_MESSAGE_DETAIL or DEFAULT_MESSAGE_DETAIL
const DEFAULT_MESSAGE_DETAIL = { ... };

...

test("does not return fileLocation, sendingOrg, and fileName", () => {
    const mockMessageDetails = {
        ...DEFAULT_MESSAGE_DETAIL,
        fileUrl: "https://azurite:10000/devstoreaccount1/receive%2Fsimple_report.csvuploader%2Fupload-covid-19-c33f9d36-9e5b-44eb-9368-218d88f3a7d1-20230131190253.csv";
    };  
});
```

The reason for this is at least twofold:
1. React state changes are triggered on _reference_ changes, not value changes.  For example, even if `MOCK_MESSAGE_DETAIL.fileUrl` is mutated, it won't register as a state change because `MOCK_MESSAGE_DETAIL` still has the same reference in memory.
2. It prevents state changes from "leaking" between test cases.  Like if writing over `MOCK_MESSAGE_DETAIL.fileUrl` [here](https://github.com/CDCgov/prime-reportstream/pull/8190/files#diff-d758df7c75cb3671680fdd77a2b4dc67c7968b61ad5428fb89bd6f923584a3aaR213) means that that's permanently changed for **all test cases** unless it's explicitly overridden in other tests.  Creating a new object for each test case makes it so that any potential mutation is fully contained in the test case.  (This increases memory complexity, but in Jest tests/internal use that's definitely tolerable.  I'd honestly also say that's pretty tolerable in React overall because lost references will just be garbage collected, and it's rare that this'll have significant performance implications unless we're implementing something in a really, really unperformant way anyway 🤷)


--- 
As a side note, you can still mutate nested fields in an object even if declared with a `const`; it's just the reference to the object that just can't change:

```js
const obj = {};
obj.a = "whatever"; // valid
obj = {}; // invalid
```

---

[Here](https://blog.logrocket.com/immutability-react-should-you-mutate-objects/)'s a good article about mutability/immutability in the context of React rendering in case you're interested!